pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -945,6 +945,17 @@ func applyUpstreamTLSSettings(opts *buildClusterOpts, tls *networking.ClientTLSS
 
 	c := opts.cluster
 	proxy := opts.proxy
+
+	// Disable transport socket when cluster type is `Cluster_ORIGINAL_DST` and mtls is autoDetected
```

Comment:
This may be more clear to set in `conditionallyConvertToIstioMtls`. Seems odd to have that function declare "auto mtls" then later override it

Reply:
make sense